web-ui/src/main/resources/catalog/components/common/map/mapService.js

```diff
@@ -841,11 +842,20 @@
               if (requestedStyle) {
                 layerParam.STYLES = requestedStyle.Name;
               } else {
-                // STYLES is mandatory parameter.
-                // ESRI will complain on this.
-                // Even &STYLES&... return an error on ESRI.
-                // TODO: Fix or workaround
-                layerParam.STYLES = '';
+                // The first style element is the default style
```

Comment:
`style` is already managed above: https://github.com/geonetwork/core-geonetwork/pull/2589/files#diff-f84568aec2e92783269ee51e149a71b4R907

Maybe you can merge, or at least put your code in the same place.

Reply:
The code you point is to populate a list of different styles available for the layer, the code here, instead, is to have the layer with a not empty style parameter (in layerParam). This is where requestedStyle is assigned to layerParam.STYLES and where, if not available, defaultStyle is needed. I think it's more easy to understand this mechanism by having the code here.

Reply:
@fgravin The code you point seem not used by OpenLayers to issue the `GetMap` request,  it's for a different purpose. For the `GetMap` request, it's required to set `layerParam.STYLES` and if it's empty as in the original code, fails for WMS ESRI layers.

I have tested the code from @PascalLike with a context file using WMS layers from ESRI service and works fine. 

Please check if looks fine so can be merged, thanks.

Reply:
It's not the same code right but you are working on same thing, the style.
The test `angular.isArray(getCapLayer.Style) && getCapLayer.Style.length > 0` is done 4 times in the same method, was wondering why the style management is so spread out.

Reply:
@fgravin I optimized the code, now the check is centralized
@josegar74 please give a look

Reply:
@fgravin, @PascalLike updated the code to extract the check in a function. If looks all fine now, please merge the PR.